core_impl/src/main/java/com/google/instrumentation/stats/StatsManagerImpl.java

```diff
@@ -113,26 +114,27 @@ public void process() {
     }
   }
 
-  private static final class CreateDistributionViewFunction
-      implements Function<DistributionViewDescriptor, View> {
+  private static final class CreateMutableDistributionViewFunction
+      implements Function<DistributionViewDescriptor, MutableView> {
     @Override
-    public View apply(DistributionViewDescriptor viewDescriptor) {
-      // TODO(songya): Create Distribution Aggregations from internal Distributions,
-      // update the start and end time.
-      return DistributionView.create(
-          viewDescriptor,
-          null,
-          Timestamp.fromMillis(System.currentTimeMillis()),
+    public MutableView apply(DistributionViewDescriptor viewDescriptor) {
+      List<MutableDistribution> distributions = new ArrayList<MutableDistribution>();
+      List<Double> bucketBoundaries =
+          viewDescriptor.getDistributionAggregationDescriptor().getBucketBoundaries();
+      distributions.add(
+          MutableDistribution.create(
+              bucketBoundaries == null ? null : BucketBoundaries.create(bucketBoundaries)));
```

Comment:
I think there should also be a TODO here saying that we should only add the MutableDistributions as we receive new combinations of tags.